instanovo/diffusion/multinomial_diffusion.py

```diff
@@ -321,9 +321,26 @@ def forward(self, x_0: torch.LongTensor, **kwargs: dict) -> torch.FloatTensor:
         t = torch.randint(0, self.time_steps - 1, (x_0.shape[0],)).to(x_0.device)
 
         # 2. Compute L_t
+        loss = self._compute_loss(t=t, x_0=x_0, **kwargs).mean()
+
+        # 4. Calculate prior KL term
         log_x_0 = torch.log(one_hot(x_0, num_classes=len(self.model.residues)))
+        final_log_probs = self.model.mixture_categorical(
+            log_x=log_x_0,
+            log_alpha=self.model.cumulative_schedule[self.time_steps - 1]
+            .unsqueeze(-1)
+            .unsqueeze(-1),
+            log_alpha_complement=self.model.cumulative_schedule_complement[self.time_steps - 1]
+            .unsqueeze(-1)
+            .unsqueeze(-1),
+        )
+        uniform_log_probs = torch.log(torch.ones_like(final_log_probs) / len(self.model.residues))
+        kl_loss = self.kl_divergence(final_log_probs, uniform_log_probs).mean()
+        return loss + kl_loss
 
+    def _compute_loss(self, x_0: torch.LongTensor, t: torch.LongTensor, **kwargs) -> torch.FloatTensor:
         # -- sample x_{t+1}
```

Comment:
```suggestion
        # 1. Sample x_{t+1}
```

and change `# 3. Calculate loss`  below to `# 2. Calculate loss`.

Reply:
Fixed in https://github.com/instadeepai/InstaNovo/pull/14/commits/464d3811d557eb5abcc033ea96f671a83f720599